modules/cart/src/Form/AddToCartForm.php

```diff
@@ -163,10 +175,35 @@ public function setFormId($form_id) {
   }
 
   /**
+   * Gets the current user.
+   *
+   * @return \Drupal\Core\Session\AccountInterface
+   *   The current user;
+   */
+  public function getCurrentUser() {
+    return $this->currentUser;
+  }
+
+  /**
+   * Gets the availability manager.
+   *
+   * @return \Drupal\commerce\AvailabilityManagerInterface
+   *   The availability manager.
+   */
+  public function getAvailabilityManager() {
+    return $this->availabilityManager;
+  }
+
+  /**
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form = parent::buildForm($form, $form_state);
+    // Add base form id for form alter hooks to identify this form.
```

Comment:
Don't form alters get the form ID passed? And users could use `hook_form_BASE_FORM_ID_alter`